samples/Mvc.FileUpload/Controllers/FileController.cs

```diff
@@ -0,0 +1,130 @@
+using Microsoft.Net.Http.Headers;
+using Mvc.FileUpload.Filters;
+
+namespace Mvc.FileUpload.Controllers
+{
+    public class FileController : Controller
+    {
+        private readonly IHostingEnvironment _hostingEnvironment;
+        private readonly ILogger<FileController> _logger;
+
+        // Get the default form options so that we can use them to set the default limits for
+        // request body data
+        private static readonly FormOptions _defaultFormOptions = new FormOptions();
+
+        public FileController(IHostingEnvironment hostingEnvironment, ILogger<FileController> logger)
+        {
+            _hostingEnvironment = hostingEnvironment;
+            _logger = logger;
+        }
+
+        [HttpGet]
+        [GenerateAntiforgeryTokenCookieForAjax]
+        public IActionResult FormToUpload()
+        {
+            return View();
+        }
+
+        // 1. We disable the form value model binding here to take control of handling potentially large files.
+        // 2. Typically antiforgery tokens are sent in request body, but since we do not want to read the request body
+        //    early, the tokens are made to be sent via headers. The antiforgery token filter first looks for tokens
+        //    in the request header and then falls back to reading the body.
+        [HttpPost]
+        [DisableFormValueModelBinding]
+        [ValidateAntiForgeryToken]
+        public async Task<IActionResult> Upload()
+        {
+            if (!MultipartRequestHelper.IsMultipartContentType(Request.ContentType))
+            {
+                return BadRequest($"Expected a multipart request, but got '{Request.ContentType}'.");
+            }
+
+            //TODO: HOW CAN I MODEL BIND ALL THIS FORM URL ENCODED DATA LATER
+
+            // Used to accumulate all the form url encoded key value pairs in the request.
+            var formAccumulator = new KeyValueAccumulator();
+
+            var boundary = MultipartRequestHelper.GetBoundary(
+                MediaTypeHeaderValue.Parse(Request.ContentType), _defaultFormOptions.MultipartBoundaryLengthLimit);
+            var reader = new MultipartReader(boundary, HttpContext.Request.Body);
+
+            var section = await reader.ReadNextSectionAsync();
+            while (section != null)
+            {
+                ContentDispositionHeaderValue contentDisposition;
+                ContentDispositionHeaderValue.TryParse(section.ContentDisposition, out contentDisposition);
+
+                if (MultipartRequestHelper.HasFileContentDisposition(contentDisposition))
+                {
+                    var name = HeaderUtilities.RemoveQuotes(contentDisposition.Name) ?? string.Empty;
+                    var fileName = HeaderUtilities.RemoveQuotes(contentDisposition.FileName) ?? string.Empty;
+
+                    // Copy the uploaded file to local disk
+                    var targetFilePath = Path.Combine(_hostingEnvironment.ContentRootPath, Guid.NewGuid().ToString());
```

Comment:
I wanted to keep the sample simple without much setup(example: setting up azure blog storage etc.) required. However, I think I could have a sample showing copying stream to a azure blob.